packages/uniforms/__tests__/BaseForm.tsx

```diff
@@ -1,387 +1,76 @@
-
-      expect(onChange).not.toBeCalled();
-    });
-  });
-
-  describe('when reset', () => {
-    const createWrapper = () =>
-      mount<BaseForm<any>>(
-        <BaseForm model={model} schema={schema} onSubmit={onSubmit} />,
-      );
-
-    it('increase `resetCount`', () => {
-      const wrapper = createWrapper();
-      wrapper.instance().reset();
-
-      expect(wrapper.state('resetCount')).toBe(1);
-    });
-
-    it('sets submitted back to false', async () => {
-      const wrapper = createWrapper();
-      const instance = wrapper.instance();
-      expect(instance.getContext().submitted).toBe(false);
-      wrapper.find('form').simulate('submit');
-      expect(instance.getContext().submitted).toBe(true);
-      instance.reset();
-      expect(instance.getContext().submitted).toBe(false);
+      expect(container.getElementsByTagName('div')).toHaveLength(3);
     });
   });
 
   describe('when submitted', () => {
-    const createWrapper = () =>
-      mount<BaseForm<any>>(
-        <BaseForm model={model} schema={schema} onSubmit={onSubmit} />,
-      );
-
     it('calls `onSubmit` once', () => {
-      const wrapper = createWrapper();
-      wrapper.find('form').simulate('submit');
+      render(
+        <BaseForm schema={schema} onSubmit={onSubmit} data-testid="form" />,
+      );
 
+      fireEvent.submit(screen.getByTestId('form'));
       expect(onSubmit).toHaveBeenCalledTimes(1);
     });
 
     it('calls `onSubmit` with correct model', () => {
-      const wrapper = createWrapper();
-      wrapper.find('form').simulate('submit');
+      render(
+        <BaseForm
+          model={model}
+          schema={schema}
+          onSubmit={onSubmit}
+          data-testid="form"
+        />,
+      );
 
+      fireEvent.submit(screen.getByTestId('form'));
       expect(onSubmit).toHaveBeenLastCalledWith(model);
     });
 
     it('calls `onSubmit` with the correctly `modelTransform`ed model', () => {
-      const wrapper = createWrapper();
-      wrapper.setProps({
-        modelTransform(mode, model) {
-          return mode === 'submit' ? { submit: 1 } : model;
-        },
-      });
-
-      wrapper.find('form').simulate('submit');
+      render(
+        <BaseForm
+          model={model}
+          schema={schema}
+          onSubmit={onSubmit}
+          data-testid="form"
+          modelTransform={(mode, model) =>
+            mode === 'submit' ? { submit: 1 } : model
+          }
+        />,
+      );
 
+      fireEvent.submit(screen.getByTestId('form'));
       expect(onSubmit).toHaveBeenLastCalledWith({ submit: 1 });
-
-      wrapper.setProps({ modelTransform: undefined });
-    });
-
-    it('sets `submitted` to true', async () => {
-      const wrapper = createWrapper();
-      const instance = wrapper.instance();
-      expect(instance.getContext().submitted).toBe(false);
-      wrapper.find('form').simulate('submit');
-      expect(instance.getContext().submitted).toBe(true);
-    });
```

Comment:
Idea: Prepare a component that reads `context.submitted` and renders conditionally.
Then we can `expect(Component).toHaveBeenCalledTimes(1)`.